### PR TITLE
Integrate Qdrant as another vector store

### PR DESCRIPTION
**Pull request by kacperlukawski**
As suggested in https://github.com/jerryjliu/gpt_index/issues/293 This PR provides Qdrant as another vector store and reader. It's still in progress.

### CODE REVIEW

**Comment by jerryjliu** [comment]
i love this, just ping me when you want me to review!

**Comment by kacperlukawski** [comment]
@jerryjliu I think it's ready for the review :)

File: gpt_index/indices/vector_store/qdrant.py [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+            collection_name = index_struct.get_collection_name()
+
+            # Create the Qdrant collection, if it does not exist yet
+            if not self._collection_initialized:
+                self._create_collection(
+                    collection_name=collection_name,
+                    vector_size=len(text_embedding),
+                )
+
+            while True:
```
Comment (jerryjliu) [inline gpt_index/indices/vector_store/qdrant.py:117]: what's the purpose of this? is there not an easier way to see if the id exists?

**Review by jerryjliu — COMMENTED** [review]
This is really impressive, congrats @kacperlukawski. I left a few comments but I'm going to go ahead and try out the code + nb examples for myself.

don't worry about linting etc. I can handle that

**Pull request merged**